lib/vector-config-macros/src/configurable_component.rs

```diff
@@ -73,11 +240,28 @@ pub fn configurable_component_impl(args: TokenStream, item: TokenStream) -> Toke
         }
     };
 
-    let component_type = options
-        .component_type()
-        .map(|ct| ct.as_str().to_string())
-        .map(|s| quote! { #[configurable(metadata(component_type = #s))] });
+    // If the component is typed -- source, transform, sink -- we do a few additional things:
+    // - we add a metadata attribute to indicate the component type
+    // - we add an attribute so the component's configuration type becomes "named", which drives
+    //   the component config trait impl (i.e. `SourceConfig`) and will eventually drive the value
+    //   that `serde` uses to deserialize the given component variant in the Big Enum model
+    // - we automatically generate the call to register the component config type via `inventory`p
+    //   which powers the `vector generate` subcommand by maintaining a name -> config type map
```

Comment:
_Potentially._

I think it's actually easier to keep `inventory`, and only `inventory`, just for the example config stuff. Otherwise, we would need another big match block type of method to go from the component name to example config... and we'd have to use _another_ procedural macro just for that if we wanted it to automatically be in sync with the enum.

Reply:
Fixed in 98f744c17.